pkg/config/remote/configs_apm_sampling.go

```diff
@@ -26,20 +26,18 @@ func newApmSamplingConfigs() *apmSamplingConfigs {
 }
 
 func (c *apmSamplingConfigs) update(configFiles map[string]configFiles) (*APMSamplingUpdate, error) {
-	if len(configFiles) > 1 {
-		return nil, fmt.Errorf("apm sampling expects one config max. %d received", len(configFiles))
+	if !c.shouldUpdate(configFiles) {
+		return nil, nil
+	}
+	update := &APMSamplingUpdate{
+		Config: &APMSamplingConfig{
+			Configs: make(map[string]Config, len(configFiles)),
+		},
 	}
 	for configID, files := range configFiles {
-		if c.config != nil && c.config.ID == configID && c.config.Version >= files.version() {
-			return nil, nil
-		}
-		update := &APMSamplingUpdate{
-			Config: &APMSamplingConfig{
-				Config: Config{
-					ID:      configID,
-					Version: files.version(),
-				},
-			},
+		update.Config.Configs[configID] = Config{
+			ID:      configID,
+			Version: files.version(),
 		}
 		for _, file := range files {
```

Comment:
This is a bit confusing. Do we have a set of files under one config id?

Reply:
Yes I know it's unexpected, some products wanted to bake in "diffs" which led to this mess. If you don't want that capability feel free to error out if there are more than one file per config.